### PR TITLE
Add vector_reproject

### PR DESCRIPTION
**Pull request by m-mohr**
Based on discussions in https://github.com/Open-EO/openeo-processes/pull/382#discussion_r1016698234

This adds a new resample_vector process.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - `flatten_dimensions`
     - `load_ml_model`
     - `predict_random_forest`
+    - `resample_vector`
```
Comment (dthiex) [inline CHANGELOG.md:18]: Sorry if I missed the discussion about this somewhere but what about naming this `reproject_vector`. Without knowing what the process does I assumed from the name something else (that it would modify the data - aggregate it but it seems to "only" change the CRS).

**Review by dthiex — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (clausmichele) [reply CHANGELOG.md:18]: I agree with @dthiex, the name does not seem to fit, since there is no resampling involved but just a reprojection of the points composing the vector data.

Comment (m-mohr) [reply CHANGELOG.md:18]: I was just taking the prefix from resample_spatial and resample_cube_temporal etc as it seems to be a group of similar processes. It changes the CRS, it may change the labels, but not the actual data. I have no strong preference either way though.

Comment (m-mohr) [reply CHANGELOG.md:18]: Rename to ` vector_reproject`.

**Comment by m-mohr** [comment]
The comment has been solved, please review (again).

This PR also includes a small change in the categories (one category "aggregate & spatial" into two categories "aggregate" and "reproject").

File: proposals/vector_reproject.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
         }
     ],
     "returns": {
-        "description": "A vector data cube with geometries converted into the new projection. The reference system of the vector dimension changes, all other dimensions and properties remain unchanged.",
+        "description": "A vector data cube with geometries converted into the new projection. The reference system of the vector dimensions changes all other dimensions and properties remain unchanged.",
```
Comment (dthiex) [inline proposals/vector_reproject.json:47]: Why did you remove the comma here? I think it's correct an makes the sentence much easier to read/understand.

**Review by dthiex — APPROVED** [review]

File: proposals/vector_reproject.json (hunk 2 above)
Comment (m-mohr) [reply proposals/vector_reproject.json:47]: Not sure either, I'll add it again.

```suggestion
        "description": "A vector data cube with geometries converted into the new projection. The reference system of the vector dimensions changes, all other dimensions and properties remain unchanged.",
```

Comment (m-mohr) [reply proposals/vector_reproject.json:47]: I'm wondering, whether having a dimension parameter as in filter_temporal makes sense?!

Comment (clausmichele) [reply proposals/vector_reproject.json:47]: @m-mohr I guess that you commented in the wrong PR?

Comment (m-mohr) [reply proposals/vector_reproject.json:47]: @clausmichele No, it's the right PR. In theory there could be two geometry dimensions, right?

Comment (clausmichele) [reply proposals/vector_reproject.json:47]: But are you talking about the `filter_temporal` process or `vector_reproject`? In the comment of 2 hours ago you mentioned `filter_temporal`.

Comment (m-mohr) [reply proposals/vector_reproject.json:47]: Yes, filter_temporal also has a dimension parameter and I mentioned that as an example for how it could be done in this process as well.

Comment (clausmichele) [reply proposals/vector_reproject.json:47]: Ok sorry, I misunderstood.
Anyway I agree, probably it could be good to have a dimension parameter with the default set to `geometries`.

Comment (m-mohr) [reply proposals/vector_reproject.json:47]: Yes, but defaults to `geometry` ;-)

Comment (m-mohr) [reply proposals/vector_reproject.json]: Added the parameter and added the comma.

File: proposals/vector_reproject.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
+{
+    "id": "vector_reproject",
+    "summary": "Reprojects the geometry dimension",
+    "description": "Converts the geometries stored in a geometry dimensions to a different projection.\n\nTo spatially align two data cubes with each other, better use the process ``resample_cube_spatial()``.",
```
Comment (LukeWeidenwalker) [inline proposals/vector_reproject.json:4]:
```suggestion
    "description": "Converts the geometries stored in a geometry dimensions to a different projection.\n\nTo spatially align two data cubes with each other, better use the process `resample_cube_spatial()`.",
```

**Review by LukeWeidenwalker — APPROVED** [review]
lgtm

File: proposals/vector_reproject.json (hunk 3 above)
Comment (m-mohr) [reply proposals/vector_reproject.json:4]: The two characters are correct as it enables the internal linking between processes.

Comment (LukeWeidenwalker) [reply proposals/vector_reproject.json:4]: Ah, apologies, thanks for clarifying! :)

Comment (m-mohr) [inline proposals/vector_reproject.json:4]:
```suggestion
    "description": "Converts the geometries stored in a geometry dimensions to a different projection.",
```

resample_cube_spatial doesn't support vector data, so the note here doesn't make sense.

File: proposals/vector_reproject.json [hunk 4]
```diff
@@ -0,0 +1,97 @@
+{
+    "id": "vector_reproject",
+    "summary": "Reprojects the geometry dimension",
+    "description": "Converts the geometries stored in a geometry dimensions to a different projection.",
```
Comment (m-mohr) [inline proposals/vector_reproject.json:4]:
```suggestion
    "description": "Converts the geometries stored in a geometry dimension to a different coordinate reference system.",
```

File: proposals/vector_reproject.json [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                "dimensions": [
+                    {
+                        "type": "geometry"
+                    }
+                ]
+            }
+        },
+        {
+            "name": "projection",
+            "description": "Projection to reproject to. specified as as [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```
Comment (m-mohr) [inline proposals/vector_reproject.json:27]:
```suggestion
            "description": "Coordinate reference system to reproject to. specified as as [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```

File: proposals/vector_reproject.json [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                    "string",
+                    "null"
+                ]
+            },
+            "default": null,
+            "optional": true
+        }
+    ],
+    "returns": {
+        "description": "A vector data cube with geometries projected to the new projection. The reference system of the geometry dimension changes, all other dimensions and properties remain unchanged.",
```
Comment (m-mohr) [inline proposals/vector_reproject.json:59]:
```suggestion
        "description": "A vector data cube with geometries projected to the new coordinate reference system. The reference system of the geometry dimension changes, all other dimensions and properties remain unchanged.",
```

File: proposals/vector_reproject.json [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                "dimensions": [
+                    {
+                        "type": "geometry"
+                    }
+                ]
+            }
+        },
+        {
+            "name": "projection",
+            "description": "Coordinate reference system to reproject to. specified as as [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```
Comment (mkadunc) [inline proposals/vector_reproject.json:27]:
```suggestion
            "description": "Coordinate reference system to reproject to. Specified as an [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```

File: proposals/vector_reproject.json [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+                "dimensions": [
+                    {
+                        "type": "geometry"
+                    }
+                ]
+            }
+        },
+        {
+            "name": "projection",
+            "description": "Coordinate reference system to reproject to. Specified as an [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```
Comment (m-mohr) [inline proposals/vector_reproject.json:27]:
```suggestion
            "description": "Coordinate reference system to reproject to. Specified as an [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), or [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
```

Comment (m-mohr) [reply proposals/vector_reproject.json:27]: Actually, I think we should remove PROJ definitions.

**Comment by m-mohr** [comment]
Ready for review
